test-utils/src/types.rs

```diff
@@ -314,3 +314,57 @@ async fn connection_task(socket: tokio::net::TcpStream, mode: ServerMode, mut ex
 		}
 	}
 }
+
```

Comment:
This file is called `types.rs` which isn't really what it is about anymore. Perhaps merge it with `helpers.rs`?